app/src/main/java/com/yacgroup/yacguide/database/tourbook/TourbookExporter.kt

```diff
@@ -20,165 +20,30 @@ package com.yacgroup.yacguide.database.tourbook
 
 import android.content.ContentResolver
 import android.net.Uri
-import com.yacgroup.yacguide.database.Ascend
 import com.yacgroup.yacguide.database.DatabaseWrapper
-import com.yacgroup.yacguide.database.Partner
-
-import com.yacgroup.yacguide.utils.ParserUtils
-import org.apache.commons.csv.CSVFormat
-import org.apache.commons.csv.CSVPrinter
-
-import org.json.JSONArray
 import org.json.JSONException
-import org.json.JSONObject
-
-import java.io.*
-import java.util.ArrayList
+import java.io.IOException
 
 class TourbookExporter(
     private val _db: DatabaseWrapper,
     private val _contentResolver: ContentResolver) {
 
-    private val JSON_INDENT_SPACE = 2
-
-    private val _routeIdKey = "routeId"
-    private val _styleIdKey = "styleId"
-    private val _yearKey = "year"
-    private val _monthKey = "month"
-    private val _dayKey = "day"
-    private val _partnersKey = "partners"
-    private val _notesKey = "notes"
-
-    var exportFormat: TourbookExportFormat = TourbookExportFormat.eJSON
+    var exportFormat = TourbookExportFormat.eJSON
 
     @Throws(IOException::class)
     fun exportTourbook(uri: Uri) {
         when (exportFormat) {
```

Comment:
Why do you check for `exportFormat` and passing it to the concrete exporter implementations, too?
E.g. the `CsvExporter` should know about it's format (and nobody else should).
- You should remove the `TourbookExportFormat` enum and move the definitions from there to the corresponding concrete implementations (like `CSVExporter` etc.).
- The `BaseExporter`s `export` API may then omit the `exportFormat` parameter.
- For `JsonExporter` you may use a boolean `isVerbose` or derive a subclass `JsonVerboseExporter` from `JsonExporter` and override the corresponding method